File: game/Assets/_Game/Scripts/Network/Schema/ExplorerSchema.cs
// Generated schema — mirrors backend/src/colyseus/schemas/ExplorerSchema.ts

using Colyseus.Schema;
#if UNITY_5_3_OR_NEWER
using UnityEngine.Scripting;
#endif

public partial class ExplorerSchema : Schema
{
#if UNITY_5_3_OR_NEWER
    [Preserve]
#endif
    public ExplorerSchema() { }

    [Type(0,  "string")]  public string explorerId   = default;
    [Type(1,  "string")]  public string playerId     = default;
    [Type(2,  "number")]  public float  x            = 0;
    [Type(3,  "number")]  public float  y            = 0;
    [Type(4,  "string")]  public string state        = default;
    [Type(5,  "number")]  public float  score        = 0;
    [Type(6,  "number")]  public float  coinCount    = 0;
    [Type(7,  "boolean")] public bool   hasBag       = false;
    [Type(8,  "boolean")] public bool   hasBoat      = false;
    [Type(9,  "boolean")] public bool   hasShield    = false;
    [Type(10, "boolean")] public bool   isBot        = false;
    [Type(11, "number")]  public float  botMoveCount = 0;
}

File: backend/src/colyseus/schemas/ExplorerSchema.ts
import { Schema, type } from '@colyseus/schema'

export class ExplorerSchema extends Schema {
  @type('string') explorerId: string = ''
  @type('string') playerId: string = ''
  @type('number') x: number = 0
  @type('number') y: number = 0
  @type('string') state: string = 'idle'
  @type('number') score: number = 0
  @type('number') coinCount: number = 0
  @type('boolean') hasBag: boolean = false
  @type('boolean') hasBoat: boolean = false
  @type('boolean') hasShield: boolean = false
  @type('boolean') isBot: boolean = false
  @type('number') botMoveCount: number = 0
}
